packages/common/src/utils/formatting.ts

```diff
@@ -97,11 +98,13 @@ const getTimeFormat = (
     return `YYYY-MM-DD, ${timeFormat} (Z)`;
 };
 
+// TODO: To rename to isDayJsInput once we remove moment usage
 export const isMomentInput = (value: unknown): value is MomentInput =>
```

Comment:
little bit afraid of this change as `isMomentInput` is used in backend/common/frontend at the same time. did you have a chance to test all 3 cases?

Reply:
I've tested the following against 2 render environments: one with and one without this change. 

FE: 
- Tooltips on hover of dates of charts 
- `Filter dashboard to` with dates on 3 different timezones: London, Madrid, and San Francisco
- `Filter chart to` with dates on 3 different timezones: London, Madrid, and San Francisco
- Scheduled deliveries' filters: Change filter of `in the last completed

coming soon with `common` report

Reply:
The `backend` doesn't use `dayjs` at the moment so this `isMomentInput` change (adding an extra OR) won't be impacted.

But I'm looking at places where `common` is using `dayjs` and see where those functions are called (along with `isMomentInput`

Reply:
Backend: 
`CsvService` uses this for getting the row value for `timestamps` and `dates` here: https://github.com/lightdash/lightdash/blob/feat/frontend-replace-moment-with-dayjs/packages/backend/src/services/CsvService/CsvService.ts#L93 and also uses `formatItemValue`, which also calls `isMomentInput`:

This PR's [render chart table with lots of date dimensions](https://lightdash-pr-9323.onrender.com/projects/3675b69e-8324-4110-bdca-059031aa8da3/tables/events?create_saved_chart_version=%7B%22tableName%22%3A%22events%22%2C%22metricQuery%22%3A%7B%22exploreName%22%3A%22%22%2C%22dimensions%22%3A%5B%22events_timestamp_ltz_raw%22%2C%22events_timestamp_ltz_day%22%2C%22events_timestamp_ntz_raw%22%2C%22events_timestamp_ntz_day%22%2C%22events_timestamp_tz_raw%22%2C%22events_timestamp_tz_millisecond%22%2C%22events_timestamp_tz_minute%22%2C%22events_timestamp_tz_second%22%2C%22events_timestamp_tz_hour%22%2C%22events_timestamp_tz_day%22%5D%2C%22metrics%22%3A%5B%5D%2C%22filters%22%3A%7B%7D%2C%22sorts%22%3A%5B%7B%22fieldId%22%3A%22events_timestamp_tz_day%22%2C%22descending%22%3Atrue%7D%5D%2C%22limit%22%3A500%2C%22tableCalculations%22%3A%5B%5D%2C%22additionalMetrics%22%3A%5B%5D%7D%2C%22tableConfig%22%3A%7B%22columnOrder%22%3A%5B%22events_timestamp_ltz_raw%22%2C%22events_timestamp_ltz_day%22%2C%22events_timestamp_ntz_raw%22%2C%22events_timestamp_ntz_day%22%2C%22events_timestamp_tz_raw%22%2C%22events_timestamp_tz_millisecond%22%2C%22events_timestamp_tz_minute%22%2C%22events_timestamp_tz_second%22%2C%22events_timestamp_tz_hour%22%2C%22events_timestamp_tz_day%22%5D%7D%2C%22chartConfig%22%3A%7B%22type%22%3A%22cartesian%22%2C%22config%22%3A%7B%22layout%22%3A%7B%22xField%22%3A%22events_timestamp_ltz_raw%22%2C%22yField%22%3A%5B%22events_timestamp_ltz_day%22%5D%7D%2C%22eChartsConfig%22%3A%7B%22series%22%3A%5B%7B%22type%22%3A%22bar%22%2C%22yAxisIndex%22%3A0%2C%22encode%22%3A%7B%22xRef%22%3A%7B%22field%22%3A%22events_timestamp_ltz_raw%22%7D%2C%22yRef%22%3A%7B%22field%22%3A%22events_timestamp_ltz_day%22%7D%7D%7D%5D%7D%7D%7D%7D)
Another [render](https://lightdash-pr-9355.onrender.com/projects/3675b69e-8324-4110-bdca-059031aa8da3/tables/events?create_saved_chart_version=%7B%22tableName%22%3A%22events%22%2C%22metricQuery%22%3A%7B%22exploreName%22%3A%22events%22%2C%22dimensions%22%3A%5B%22events_timestamp_ltz_raw%22%2C%22events_timestamp_ltz_day%22%2C%22events_timestamp_ntz_raw%22%2C%22events_timestamp_ntz_day%22%2C%22events_timestamp_tz_raw%22%2C%22events_timestamp_tz_millisecond%22%2C%22events_timestamp_tz_minute%22%2C%22events_timestamp_tz_second%22%2C%22events_timestamp_tz_hour%22%2C%22events_timestamp_tz_day%22%5D%2C%22metrics%22%3A%5B%5D%2C%22filters%22%3A%7B%7D%2C%22sorts%22%3A%5B%7B%22fieldId%22%3A%22events_timestamp_tz_day%22%2C%22descending%22%3Atrue%7D%5D%2C%22limit%22%3A500%2C%22tableCalculations%22%3A%5B%5D%2C%22additionalMetrics%22%3A%5B%5D%7D%2C%22tableConfig%22%3A%7B%22columnOrder%22%3A%5B%22events_timestamp_ltz_raw%22%2C%22events_timestamp_ltz_day%22%2C%22events_timestamp_ntz_raw%22%2C%22events_timestamp_ntz_day%22%2C%22events_timestamp_tz_raw%22%2C%22events_timestamp_tz_millisecond%22%2C%22events_timestamp_tz_minute%22%2C%22events_timestamp_tz_second%22%2C%22events_timestamp_tz_hour%22%2C%22events_timestamp_tz_day%22%5D%7D%2C%22chartConfig%22%3A%7B%22type%22%3A%22cartesian%22%2C%22config%22%3A%7B%22layout%22%3A%7B%22xField%22%3A%22events_timestamp_ltz_raw%22%2C%22yField%22%3A%5B%22events_timestamp_ltz_day%22%5D%7D%2C%22eChartsConfig%22%3A%7B%22series%22%3A%5B%7B%22type%22%3A%22bar%22%2C%22yAxisIndex%22%3A0%2C%22encode%22%3A%7B%22xRef%22%3A%7B%22field%22%3A%22events_timestamp_ltz_raw%22%7D%2C%22yRef%22%3A%7B%22field%22%3A%22events_timestamp_ltz_day%22%7D%7D%7D%5D%7D%7D%7D%7D)

Tested downloading the csv of these results in 2 ways:
- Formatted
- Raw
 and saw no diffs between the two (used online diff check tool)
 
 
 `formatItemValue` here https://github.com/lightdash/lightdash/blob/feat/frontend-replace-moment-with-dayjs/packages/common/src/utils/formatting.ts#L365
 - we have the timezone tests
 - `formatRows` on the BE: https://github.com/lightdash/lightdash/blob/feat/frontend-replace-moment-with-dayjs/packages/common/src/index.ts#L911 - tested this by looking at the rows in table vizualisations + on tooltips and line charts, and dashboard charts with the `filter dashboard to` and dates and consistent between render environments

Tested `formatRows`: 
- On charts
before
<img width="417" alt="Screenshot 2024-03-13 at 17 45 11" src="https://github.com/lightdash/lightdash/assets/7611706/664a1144-9ffb-4f50-ac9c-b0f9793cf6c2">
after
<img width="440" alt="Screenshot 2024-03-13 at 17 43 52" src="https://github.com/lightdash/lightdash/assets/7611706/c3321307-55ef-439a-9243-7bc507f6fb46">
- On charts in dashboard
before
<img width="566" alt="Screenshot 2024-03-13 at 17 45 36" src="https://github.com/lightdash/lightdash/assets/7611706/60d48fd7-0831-459a-a7e3-1d0f354b25f2">
after
<img width="692" alt="Screenshot 2024-03-13 at 17 45 47" src="https://github.com/lightdash/lightdash/assets/7611706/37a9355f-5f7d-4d0c-95f6-79dfd9d25aa4">
- On underlying data
 
before
<img width="1725" alt="Screenshot 2024-03-13 at 17 52 21" src="https://github.com/lightdash/lightdash/assets/7611706/bfb11bf5-1c5e-4e77-9165-1888d79ba7a2">
after
<img width="1364" alt="Screenshot 2024-03-13 at 17 52 28" src="https://github.com/lightdash/lightdash/assets/7611706/51df17a9-f560-4762-be03-b4abacf59583">

Reply:
Also added a simple test here: https://github.com/lightdash/lightdash/pull/9323/commits/87648914f2390d7e94670c4abafee90f94fc0280